src/main/types/SIOP.types.ts

```diff
@@ -108,39 +134,42 @@ export interface AuthenticationResponseOpts {
   did: string;
   vp?: VerifiablePresentationResponseOpts[];
   expiresIn?: number;
+  accessToken?: string;
+  tokenType?: string;
+  refreshToken?: string;
+}
+
+export interface IdToken extends JWTPayload {
+  iss?: ResponseIss.SELF_ISSUED_V2 | string;
+  sub?: string; // did (or thumbprint of sub_jwk key when type is jkt)
+  aud?: string; // redirect_uri from request
+  iat?: number; // Issued at time
+  exp?: number; // Expiration time
+  auth_time?: number;
+  nonce?: string;
+  _vp_token?: {
+    /*
+      This profile currently supports including only a single VP in the VP Token.
+      In such cases, as defined in section 5.2 of OpenID4VP ID1, when the Self-Issued OP returns a single VP in the vp_token,
+      VP Token is not an array, and a single VP is passed as a vp_token. In this case, the descriptor map would contain a simple path expression “$”.
+      * It's not clear from the ID1 specs how to handle presentation submission in case of multiple VPs
+    */
+    presentation_submission?: PresentationSubmission;
+  };
+}
+
+export interface AuthenticationResponsePayload {
+  access_token?: ResponseIss.SELF_ISSUED_V2 | string;
+  token_type?: string;
+  refresh_token?: string;
+  expires_in: number;
+  state: string;
+  id_token: string;
+  vp_token: VerifiablePresentationPayload[] | VerifiablePresentationPayload;
```

Comment:
This is what the specs say: "A vp_token MUST be provided in the same response as the id_token of the respective OpenID Connect transaction. Depending on the response/grant type, this can be either the authentication response or the token response.The vp_token either contains a single verifiable presentation or an array of verifiable presentations...". What am I missing?

Reply:
You are missing all of https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#section-6.1
Also the fact that this was in the original code and you removed it was a good indication you needed to ask for clarification first

Reply:
Fixed